src/site/stages/build/plugins/process-entry-names.js

```diff
@@ -0,0 +1,103 @@
+  // TeamSite pages such as "benefits.va.gov" have hardcoded references to certain
+  // JavaScript bundles so that we can inject our header/footer into pages outside
+  // of our source. Therefore, we can't apply a hash to those bundles.
+  const teamsiteInjectionBundles = {
+    'proxy-rewrite.js': 'generated/proxy-rewrite.entry.js',
+    'vendor.js': 'generated/vendor.entry.js',
+    'polyfills.js': 'generated/polyfills.entry.js',
+    'styleConsolidated.css': 'generated/styleConsolidated.css',
+    'static-pages.css': 'generated/static-pages.css',
+  };
+
+  for (const fileManifestKey of Object.keys(teamsiteInjectionBundles)) {
+    const teamSitePath = teamsiteInjectionBundles[fileManifestKey];
+    const hashedFileName = entryNamesDictionary.get(fileManifestKey);
+
+    if (!hashedFileName) {
+      if (buildOptions.entry) {
+        // A command arg was passed to build only certain areas of the site, so
+        // it makes that it would be missing.
+        continue;
+      } else {
+        throw new Error('Missing Webpack assets required for TeamSite!');
+      }
+    }
+
+    const hashedFileNameWithoutLeadingSlash = hashedFileName.slice(1);
+    const file = files[hashedFileNameWithoutLeadingSlash];
+
+    if (!file) continue;
+
+    files[teamSitePath] = file;
+  }
+}
+
+function getEntryNamesDictionary(buildOptions, files) {
+  const isDevBuild = [environments.LOCALHOST, environments.VAGOVDEV].includes(
+    buildOptions.buildtype,
+  );
+
+  if (isDevBuild) {
+    return {
+      get(entryName) {
+        const isJs = path.extname(entryName) === '.js';
+        const fileName = isJs
+          ? `${path.parse(entryName).name}.entry.js`
+          : entryName;
+        return `/generated/${fileName}`;
+      },
+    };
+  }
+
+  const fileManifest = files[FILE_MANIFEST_FILENAME];
+
+  if (!fileManifest) {
+    throw new Error('file-manifest.json (generated by Webpack) not found!');
+  }
+
+  const entryNamesDictionary = new Map(
+    Object.entries(JSON.parse(fileManifest.contents)),
+  );
+
+  return entryNamesDictionary;
+}
+
+function processEntryNames(buildOptions) {
+  return (files, metalsmith, done) => {
+    const entryNamesDictionary = getEntryNamesDictionary(buildOptions, files);
+
+    for (const fileName of Object.keys(files)) {
+      const file = files[fileName];
+
+      const { dom } = file;
+      if (!dom) continue;
+
+      dom('script[data-entry-name],link[data-entry-name]').each((index, el) => {
+        const $el = dom(el);
+        const entryName = $el.data('entryName');
+        const attribute = $el.is('script') ? 'src' : 'href';
+        const hashedEntryName = entryNamesDictionary.get(entryName);
+
+        $el.attr(attribute, hashedEntryName);
+
+        file.modified = true;
+      });
+    }
+
+    if (!buildOptions.isPreviewServer) {
+      copyAssetsToTeamSitePaths(buildOptions, files, entryNamesDictionary);
+    }
+
+    done();
+  };
+}
+
+module.exports = processEntryNames;
```

Comment:
I renamed the `addAssetHashes` plugin to `processEntryNames` because -. 
- I wanted to also apply it to local/dev builds, which don't use hashes but do change `.js` to `.entry.js`
- By enabling it for local/dev, we can always generated the asset path for the `script src` or `link href` attributes and not have to spec. This way, if there's an issue with the plugin, we may spot it on local as opposed to a higher env.
- Increased visibility into what this plugin does

Reply:
:100: I dig it!